src/main/java/uk/gov/hmcts/reform/sscs/config/VerifyTokenService.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.hmcts.reform.sscs.config;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSVerifier;
+import com.nimbusds.jose.crypto.factories.DefaultJWSVerifierFactory;
+import com.nimbusds.jose.jwk.AsymmetricJWK;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.SecretJWK;
+import com.nimbusds.jose.proc.JWSVerifierFactory;
+import com.nimbusds.jwt.SignedJWT;
+import java.net.URL;
+import java.security.Key;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+
+@Service
+@Slf4j
+public class VerifyTokenService {
+
+    @Value("${idam.jwkurl}")
+    private String idamJwkUrl;
+
+    private final JWSVerifierFactory jwsVerifierFactory;
+
+    public VerifyTokenService() {
+        this.jwsVerifierFactory = new DefaultJWSVerifierFactory();
+    }
+
+    public boolean verifyTokenSignature(String token) {
+        try {
+            SignedJWT signedJwt = SignedJWT.parse(token);
+
+            JWKSet jsonWebKeySet = loadJsonWebKeySet(idamJwkUrl);
+
+            JWSHeader jwsHeader = signedJwt.getHeader();
+            Key key = findKeyById(jsonWebKeySet, jwsHeader.getKeyID());
+
+            JWSVerifier jwsVerifier = jwsVerifierFactory.createJWSVerifier(jwsHeader, key);
+
+            return signedJwt.verify(jwsVerifier);
+        } catch (Exception e) {
+            log.error("Token validation error {}", e);
```

Comment:
log.error used to throw exceptions in AppInsight. We might want to  check with Thor or Hem if we want to get exceptions in AppInsight when this happens.